src/main/java/com/faforever/client/config/AsyncConfig.java

```diff
@@ -46,9 +47,18 @@ public ExecutorService taskExecutor() {
   public TaskScheduler taskScheduler() {
     return new ThreadPoolTaskScheduler();
   }
-
-  @Override
-  public void destroy() {
-    taskExecutor().shutdownNow();
+  
+  @Bean
+  public DestructionAwareBeanPostProcessor threadPoolShutdownProcessor() {
```

Comment:
Maybe some logging in this method?

Reply:
ha, i had logging and thought "meh, who cares about shutdown" so i removed it xD
Can put it back in of course.